### ISSUE TITLE
Need non-variadic Swift race and join methods

### ISSUE THREAD

**Issue by nevans** [mxcl/PromiseKit#326]
As far as I can tell, Swift's type system won't let me apply arrays onto variadic parameters.

````swift
public func race<T>(promises: Promise<T>...) -> Promise<T> {
    return Promise(sealant: { resolve in
        for promise in promises {
            promise.pipe(resolve)
        }
    })
}
````
could perhaps be rewritten as
````swift
public func race<T>(promises: Promise<T>...) -> Promise<T> {
    return race(promises)
}

public func race<T>(promises: [Promise<T>]) -> Promise<T> {
    return Promise(sealant: { resolve in
        for promise in promises {
            promise.pipe(resolve)
        }
    })
}

````

**Comment by nevans** [comment]
also for `join`

**Comment by tmandry** [comment]
+1

**Comment by nathanhosselton** [comment]
Seems fine. PR welcome.

**Comment by randomiser** [comment]
Can we also have the join changed so that it can be chained along with the other functions like "then".

ie:

firstly {

}.then {

}.join {

}.always {

}.error {

}

**Comment by nathanhosselton** [comment]
Resolved by #374

**Comment by nathanhosselton** [comment]
@randomiser I think this topic probably deserves it's own discussion thread. If you would like to make a proposal of sorts we can definitely consider such a feature.
